contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,444 @@
+    function addPCVDeposit(address newPCVDeposit, uint256 weight) external override onlyGovernorOrAdmin {
+        require(IPCVDeposit(newPCVDeposit).balanceReportedIn() == token, "Deposit token must be the same as for this aggregator.");
+
+        _addPCVDeposit(newPCVDeposit, weight);
+    }
+
+    /// @notice replaces this contract with a new PCV Deposit Aggregator on the rewardsAssetManager
+    /// @param newAggregator the address of the new PCV Deposit Aggregator
+    function setNewAggregator(address newAggregator) external override onlyGovernor {
+        require(PCVDepositAggregator(newAggregator).token() == token, "New aggregator must be for the same token as the existing.");
+
+        // Add each pcvDeposit to the new aggregator
+        for (uint256 i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            if (!(IPCVDepositAggregator(newAggregator).hasPCVDeposit(pcvDepositAddress))) {
+                uint256 pcvDepositWeight = pcvDepositWeights[pcvDepositAddress];
+                IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+            }
+        }
+
+        // Send old aggregator assets over to the new aggregator
+        IERC20(token).safeTransfer(newAggregator, balance());
+
+        // No need to remove all deposits, this is a lot of extra gas.
+
+        // Finally, set the new aggregator on the rewards asset manager itself
+        IRewardsAssetManager(assetManager).setNewAggregator(newAggregator);
+
+        emit AggregatorUpdate(address(this), newAggregator);
+    }
+
+    /// @notice sets the rewards asset manager
+    /// @param newAssetManager the address of the new rewards asset manager
+    function setAssetManager(address newAssetManager) external override onlyGovernor {
+        require(newAssetManager != address(0x0), "New asset manager cannot be 0x0");
+        require(IRewardsAssetManager(newAssetManager).getToken() == token, "New asset manager must be for the same token as the existing.");
+
+        address oldAssetManager = assetManager;
+        assetManager = newAssetManager;
+
+        emit AssetManagerUpdate(oldAssetManager, newAssetManager);
+    }
+
+    // ---------- View Functions ---------------
+
+    /// @notice returns true if the given address is a PCV Deposit in this aggregator
+    /// @param pcvDeposit the address of the PCV deposit to check
+    /// @return true if the given address is a PCV Deposit in this aggregator
+    function hasPCVDeposit(address pcvDeposit) public view override returns (bool) {
+        return pcvDepositAddresses.contains(pcvDeposit);
+    }
+
+    function resistantBalanceAndFei() public view override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    function balanceReportedIn() external view override returns (address) {
```

Comment:
Natspec

Reply:
already in the interface (which isn't even IPCVGuardian)

Reply:
right, reason I didn't add this is because there's no natespec for the pcv deposit interface or pcv deposit implementation to draw from

Reply:
added